R/transmute.R

```diff
@@ -1,138 +1,56 @@
-#' chondro %>%
-#'   transmute (y, x)
+#' laser %>%
+#'   transmute (t, filename)
 #'   head # => results in a data frame
-#' chondro %>%
-#'   transmute (y, x, filename, spc = spc*2)
-#'   head # => results in a hyperSpec object
-#' flu %>%
-#'   transmute (spc, filename, c)
-#'   head # => results in a hyperSpec object
-#' flu %>%
-#'   transmute (filename, c)
-#'   head # => results in a data frame
-#' flu %>%
-#'   transmute (c=c*2, c=c*0)
-#'   # => results in a data frame
-transmute.hyperSpec <- function(.data, ...){
+#' laser %>%
+#'    transmute (-spc) # => results in a hyperSpec object
+#' laser %>%
+#'    transmute (spc2 = spc*2) %>%
+#'    transmute (spc2) %>%
+#'    transmute (spc2*2) # => results in a hyperSpec object
+transmute.hyperSpec <- function(.data, ...) {
+
   # Check if user passed in a hyperSpec object
   chk.hy(.data)
-  # Collect function arguments
-  args <- rlang::enquos(...)
-  args_names <- names(args)
-  if(length(args) == 0L){
-    return(NULL)
-  }
-  # Prepare a copy of the original hyperSpec object
-  tmp_hy <- .data
-  cols2get <- vector() # create a list to save the column names to
-  tmp_spc <- tmp_hy@data[c('spc')] # store original $spc column
-  # Prepare function arguments for transmute()
-  # assumption: the variable name and expr
-  # share the same index (i.e., args[i] is the expr for the variable names(args[i]))
-  for(i in seq_along(args)){
-    expr <- quo_name(quo_get_expr(args[[i]]))
-    # Process arguments with no names (assignments)
-    if('' %in% args_names[i]){
-      cols2get <- c(cols2get, expr)
-    # Process `spc` argument assignments
-    # Manipulate `spc` column before passing it on to transmute()
-    }else if('spc' %in% args_names[i]){
-      if(grepl('spc', expr)){
-        tmp_hy@data[c('spc')] <- tmp_spc
-        eval(parse(text = paste("tmp_hy@data[c('spc')]<-", "tmp_hy@data$", expr)))
-        if(!'spc' %in% cols2get){
-          cols2get <- c(cols2get, 'spc') # ensures there is, and only one `spc` column
-        }
-      }else{
-        # Throw an error
-        stop("$spc must be mutated from a $spc column")
-      }
-    # Process non `spc` argument assignments
-    }else{
-      assign <- paste(args_names[i],'=', expr, sep='')
-      cols2get <- c(cols2get, assign)
-    }
-  }
-  # Hand off columns (i.e., prepared arguments) to transmute()
-  transmute_args <- paste(cols2get, collapse=", ")
-  res <- eval(parse(text = paste("transmute(tmp_hy@data,", transmute_args, ")")))
+
+  # Pass transmute arguments to dplyr::transmute
+  res <- transmute(.data@data, ...)
+
   # Update labels
-  hy_update_labels(.data, res)
+  setLabels.select(.data, res)
 }
 
-# Begin unit testing (UT)
-.test(transmute.hyperSpec) <- function(){
+.test(transmute.hyperSpec) <- function() {
   context("transmute.hyperSpec")
 
-  # UT1
   test_that("non hyperSpec objects are rejected", {
     df <- data.frame(a = NA, b = NA)
     expect_error(transmute.hyperSpec(df))
   })
 
-  # UT2
```

Comment:
@eoduniyi, why did you remove the unit tests?

Reply:
UT2 was:
```R
  # UT2
  test_that("arguments are correctly passed onto mutate/transmute", {
    df <- flu
    df <- dplyr::transmute(df@data, c, c = 0, c = 1)
    expect_equivalent(transmute.hyperSpec(flu, c, c = 0, c = 1), df)
  })
```

But making sure arguments are passed in their correct order is behaviour that is the responsibility of `dplyr`

Reply:
Sure, but we still want to check that we get the correct result. We *could* have a mistake that would return an invalid hyperSpec object, or the wrong content e.g. if our code doesn't hand over the correct result form dplyr.